File: Matlab/graph6.m
function [ output_args ] = graph6( s )
    v = s(1:1) - 63;
    e = (v * (v-1)) / 2; 

    toAddSize = 6;
		
	graph = zeros(v,v);
	count = 0;
	index = 2;
		
	x = 2;
	y = 1;
		
	while count < e 
		c = s(index:index) - 63;
        disp(c);
		toAdd = zeros(1, toAddSize);
		for i = toAddSize:-1:1
            toAdd(i) = bitand(c, 1);
			c = bitshift(c, -1);
        end
        disp(toAdd);
		for i = 1:toAddSize
			if x == y
				graph(x,y) = 0;
				x = x + 1;
				y = 1;
				if x == v+1
                    output_args = graph;
					return;
                end
            end
			graph(x,y) = toAdd(i);
			graph(y,x) = toAdd(i);
			count = count + 1;
			y = y + 1;
        end
		index = index + 1;
    end
    
	output_args = graph;
end

File: Cardinality Analysis/Java/Verification/src/Verification/Graph6.java
package Verification;

import java.util.Arrays;

public class Graph6 {

	/*
	 * Converts a string in Graph6 format to an adjacency
	 * matrix for manipulation.
	 * 1) Remove the first character from s (it tells you v), calculate e.
	 * 2) For each character, subtract 63, convert into binary string
	 * 3) The bitstring remaining is the upper triangle of
	 *    the adjacency matrix.  Fill these values in to a new adjacency
	 *    matrix. Return the resulting graph.
	 */
	public static int[][] retrieveGraph(String s){
		int v = (int) (s.charAt(0)) - 63;
		int e = (v * (v-1))/2;

		int toAddSize = 6;
		
		int[][] graph = new int[v][v];
		int count = 0;
		int index = 1;
		
		int x = 1;
		int y = 0;
		
		while (count < e){
			int c = (int) (s.charAt(index)) - 63;
			int[] toAdd = new int[toAddSize];
			for(int i = toAddSize-1; i >= 0; i--){
				toAdd[i] = c & 1;
				c = c >> 1;
			}
			for (int i = 0; i < toAddSize; i++){
				if (x == y){
					graph[x][y] = 0;
					x = x+1;
					y = 0;
					if (x == v){
						return graph;
					}
				}
				graph[x][y] = toAdd[i];
				graph[y][x] = toAdd[i];
				count++;
				y++;
			}
			index++;
		}
		return graph;
	}
	
	public static void main(String[] args){
		System.out.println(Arrays.deepToString(retrieveGraph("HCfvvzz")));
	}
}
